### ISSUE TITLE
prov-app fails when "tests" is provided as component id

### ISSUE THREAD

**Issue by georgfedermann** [opendevstack/ods-provisioning-app#620]
**Describe the bug**

**To Reproduce**
1. Provision a project
2. Create a component and enter "tests" as component id
3. The Jenkins build for the newly created component will fail

**Expected behavior**
The prov-app UI should show a warning if an invalid component id is provided.

**Affected version (please complete the following information):**
 - OpenShift: 3.11
 - OpenDevStack 3.x

**Log Output Jenkins**
```
ERROR: Target directory 'tests' must not exist yet
Finished: FAILURE
```

**Comment by felipecruz91** [comment]
The [current regular expression](https://github.com/opendevstack/ods-provisioning-app/blob/4a36296b0e7f729586577d586d9884ca1802e096/src/main/java/org/opendevstack/provision/model/OpenProjectDataValidator.java#L36) that validates the component id allows `tests` to be a valid value.

If that word is a reserved keyword and therefore it shall not be used when provisioning components, then the regular expression needs to be updated accordingly to ignore it: `^(?!tests)(([A-Za-z][-A-Za-z]*)?[A-Za-z])?$`

![image](https://user-images.githubusercontent.com/15997951/98692117-0eb4bd00-236f-11eb-8e34-72bd70c56e79.png)

**Comment by clemensutschig** [comment]
@netzartist want to pick this up for 4?

**Comment by stitakis** [comment]
Server side validation is needed as well. I'll do the corresponding change in the BE.

**Comment by netzartist** [comment]
I'd suggest we create a list of reserved keywords and adapt the Regex in BE & FE accordingly incl. the error messages.

**Comment by stitakis** [comment]
implemented in #624
